### PR TITLE
fix: [CDS-71344]: Passed branch from storemetadata in validate-inputset call

### PR DESCRIPTION
**Pull request by sonaligoyal-harness**
### Summary

When remote template was made in non-default branch, and branch if not present in query param then reconcile api failed because it used to pick default branch of that repo.
Passed branch from storemetadata to fix this.

#### Screenshots

earlier:

https://github.com/harness/harness-core-ui/assets/98508399/8c75fe05-ebe6-4d6c-a391-abea0640afaa

now:

https://github.com/harness/harness-core-ui/assets/98508399/2987b0a9-829e-4016-959d-7a25e5f7998a

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

**Review by litvinovstas — APPROVED** [review]

**Review by pankajsingh-harness — APPROVED** [review]

**Review by samarth-sharma31 — APPROVED** [review]

**Comment by sonaligoyal-harness** [comment]
trigger rebase

**Review by Eric761 — APPROVED** [review]

**Comment by sonaligoyal-harness** [comment]
coverage is failing because of existing files low coverage.

File: src/modules/72-templates-library/components/TemplateStudio/TemplateStudioInternal.tsx [hunk 1]
```diff
@@ -309,7 +309,7 @@ export function TemplateStudioInternal(): React.ReactElement {
         orgIdentifier,
         templateIdentifier,
         repoIdentifier,
-        branch,
+        branch: defaultTo(storeMetadata?.branch, branch),
```
Comment (sunnykesh-harness) [inline src/modules/72-templates-library/components/TemplateStudio/TemplateStudioInternal.tsx:312]: @sonaligoyal-harness Is it tested  for different repo use case? Other than that it looks good.

Comment (sonaligoyal-harness) [reply src/modules/72-templates-library/components/TemplateStudio/TemplateStudioInternal.tsx:312]: yes, it is working.

**Pull request merged**